os/mm/mm_heap/mm_initialize.c

```diff
@@ -146,13 +146,79 @@ void mm_addregion(FAR struct mm_heap_s *heap, FAR void *heapstart, size_t heapsi
 	heapinfo_update_node((FAR struct mm_allocnode_s *)heap->mm_heapstart[IDX], 0xDEAD);
 #endif
 
+#ifdef CONFIG_MM_REALTIME_SUPPORT
+	int i = 0;
+	int j;
+	int seg_size = MM_REALTIME_SUPPORT_MIN_SIZE; /* minimum size for realtime support */
+	int mem_pool_size = 0;
+	int prev_size;
+	char *numbers = CONFIG_MM_REALTIME_SUPPORT_NUMS;
+	char *ptr;
+
+	while (*numbers != '\0') {
+		heap->mm_realtime_num[i] = (int)strtol(numbers, &ptr, 10);
+		while (*ptr == ',' || *ptr == ' ') {
+			++ptr;
+		}
+		numbers = ptr;
+		if (++i >= MM_REALTIME_SUPPORT_NUMOF_SIZES) {
+			break;
+		}
+	}
+
+	for (i = 0; i < MM_REALTIME_SUPPORT_NUMOF_SIZES; ++i) {
+		if (heap->mm_realtime_num[i] < 1) {
+			heap->mm_realtime_num[i] = 0;
```

Comment:
How about use unsigned int?

Reply:
I don't think so because it's not a size but a signed number.

Reply:
You mean it allowed negative value?

Reply:
No, but I checked whether the input is negative at line 170.
However, if this var is unsigned, it might be greater than the maximum number.

Reply:
I mentioned unsigned type because it is stable for kind of size of buffer, address, length, and so on.
We usually use size_t or ssize_t for it, That's why I commented it, choice is yours.